internal/testing/cluster.go

```diff
@@ -375,7 +375,16 @@ func (c *TestCluster) WaitForContracts() ([]api.Contract, error) {
 		return nil, err
 	}
 
-	return c.Worker.ActiveContracts()
+	contracts, err := c.Bus.ActiveContracts()
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.Worker.Contracts(contracts, time.Minute)
```

Comment:
I had to introduce a request object to pass the timeout in a way that's consistent with the rhp scan route. I personally would pass timeouts through query string parameters for instance. I chose to add contracts to the request object because I think it makes more sense from an api perspective. The route is just way more flexible if you allow to specify what contracts you want to fetch...

I still think that makes sense but don't mind changing it back if you feel differently.